django/website/contacts/tests/test_views_activation.py

```diff
@@ -0,0 +1,137 @@
+def test_reset_password_view_sends_email_when_form_valid():
+    user = User.objects.create(business_email='test@example.com')
+    form = ContactPasswordResetForm(data={'email': 'test@example.com'})
+    form.users_cache = [user]
+
+    request = RequestFactory().get('/')
+
+    mail.outbox = []
+
+    view = ResetPassword()
+    view.request = request
+    view.form_valid(form)
+
+    assert len(mail.outbox) > 0
+
+
+@pytest.mark.django_db
+def test_reset_password_view_displays_message_when_form_valid():
+    user = User.objects.create(business_email='test@example.com')
+    form = ContactPasswordResetForm(data={'email': 'test@example.com'})
+    form.users_cache = [user]
+
+    request = RequestFactory().get('/')
+    SessionMiddleware().process_request(request)
+    MessageMiddleware().process_request(request)
+
+    view = ResetPassword()
+    view.request = request
+    view.form_valid(form)
+
+    assert 'success' == request._messages._queued_messages[0].tags
+    assert 'Reset password email was sent to this contact. Please check your mailbox for further instructions.' == request._messages._queued_messages[0].message
+
+
+@pytest.mark.django_db
+@patch('contacts.views.activation.messages', new=Mock())
+def test_reset_password_view_redirects_on_success():
+    user = User.objects.create(business_email='test@example.com')
+    form = ContactPasswordResetForm(data={'email': 'test@example.com'})
+    form.users_cache = [user]
+
+    request = RequestFactory().get('/')
+
+    view = ResetPassword()
+    view.request = request
+    response = view.form_valid(form)
+
+    assert isinstance(response, HttpResponseRedirect)
+    assert reverse('login') == response.url
+
+
+@pytest.mark.django_db
+def test_reset_password_view_displays_error_message_when_form_invalid():
+    form = ContactPasswordResetForm()
+    request = RequestFactory().get('/')
+    SessionMiddleware().process_request(request)
+    MessageMiddleware().process_request(request)
+
+    view = ResetPassword()
+    view.request = request
+    view.form_invalid(form)
+
+    assert 'error' == request._messages._queued_messages[0].tags
+    assert 'Email could not be sent. Check if provided email is correct.' == request._messages._queued_messages[0].message
+
+
+@pytest.mark.django_db
+def test_password_change_view_redirects_to_personal_edit_on_success():
+    user = User.objects.create(business_email='test@example.com')
+    user.set_password('password')
+    user.save()
+
+    request = RequestFactory().post(
+        '/',
+        data={
+            'old_password': 'password',
+            'new_password1': 'new_pass',
+            'new_password2': 'new_pass',
+    })
+    request.user = user
+    request.csrf_processing_done = True
+
+    response = change_password(request)
+
+    assert isinstance(response, HttpResponseRedirect)
+    assert reverse('personal_edit') == response.url
+
+
+@pytest.mark.django_db
+def test_email_acitvation_view_shows_error_on_invalid_form():
```

Comment:
I think the title of the test doesn't describe what the test does (but the commit message where that change was introduced did!). This is testing that the email activation shows error when the user is inactive.